### PR TITLE
Allows change listener to break out of Wait loop

### LINKED ISSUE #1329: ChangeWaiters failing to release when changes feed closed
In some scenarios, change waiters aren't being release when the changes feed terminates, resulting in accumulation of goroutines (and corresponding memory usage) by Sync Gateway.  

Profiling results here, from a test of 14K concurrent continuous pull replications:
https://s3.amazonaws.com/uploads.hipchat.com/55340/2472631/qSve3NURqOPZ4Sq/goroutine.pdf
https://s3.amazonaws.com/uploads.hipchat.com/55340/2472631/rV5nSCIidRIULI7/goroutine.text

### PR DESCRIPTION
**Pull request by ajres**
fixes #1329

### CODE REVIEW

**Comment by ajres** [comment]
This is a clean commit to master of the changes applied to branch hotfixes/1.1.1.2170

File: src/github.com/couchbase/sync_gateway/rest/changes_api.go [hunk 1]
```diff
@@ -188,6 +188,7 @@ func (h *handler) sendSimpleChanges(channels base.Set, options db.ChangesOptions
 		}
 
 		encoder := json.NewEncoder(h.response)
+		var forceClose bool = false
```
Comment (tleyden) [inline src/github.com/couchbase/sync_gateway/rest/changes_api.go]: if `forceClose := false` works here, I think it should be used.  It's more concise and more commonly seen in Go code

**Comment by ajres** [comment]
The hotfix branch has been renamed to feature/hotfix_1.1.1.2170

This PR is currently behind the hotfix branch, one the hotfix has been reviewed latest changes will b merged back to this PR for merge to master.

**Comment by ajres** [comment]
This branch is now alligned with the hotfix branch feature/hotfix_1.1.1.2170

**Pull request merged**